app/misc/authenticate.ts

```diff
@@ -535,3 +535,14 @@ function getUsername(){
   // Return the users username
   return account.login;
 }
+
+// Get a git signature for the account that is logged in.
+function getSignature(){
+  // Set email to the email in account object
+  var email = account.email;
```

Comment:
Probably need to check for account (in the case when I am not logged in)?

![image](https://user-images.githubusercontent.com/15122789/59452039-b9fbc400-8dc1-11e9-9f7f-9399a6d310db.png)

Reply:
Good catch, did this happen when you differed logging in?

Reply:
Yes. "continue without sign in" then stash some changes.

Reply:
I updated the get signature function to take the repository as a parameter and return the default signature if the user has not logged in. Give it a shot and let me know if it solves this issue! :)

Reply:
Is '@users.noreply.github.com' the only type of email that will appear?

Reply:
If the account has a public email address, then that public email address will be applied to the signature, if not the email will follow the format of `<uesrname>@users.noreply.github.com`.